### ISSUE TITLE
Unmet dependencies Installing Standalone iVozProvider

### ISSUE THREAD

**Issue by chiragd** [irontec/ivozprovider#39]
Hi all,

I think iVozProvider looks great and I can't wait to test it out. I'm having trouble installing on ubuntu.

I've updated the sources but keep getting told that the ivozprovider has unmet dependencies. When I try to install manually I'm still getting stuck. Running Ubuntu 16 LTS.

Would appreciate if you could point me in the right direction.

Thanks,

C

**Comment by Kaian** [comment]
Hi @chiragd 

Sorry for the late response. I think you send your request in our web, but github issue tracker is the best way to get feedback from devs :)

That was totally my fault, the missing dependency was sngrep, a capturing tool we also develop and due to an error in the repository, I deleted the packages without checking the consistency of the installation of ivozprovider. I have already added the package so the problem should be fixed now.

We have only tested ivozprovider debian packages under Debian Jessie 8. With ubuntu it may or may not work.

Thanks for testing ivozprovider and reporting the problem!!

**Comment by Kaian** [comment]
I have tested in a docker container with ubuntu xenial and I cant ensure that it won't work, because package names from debian are different from ubuntu and dependencies are not fulfilled.

For example, ivozporivder-data-library depends on php5-cli while ubuntu only has php7.0-cli, so I highly recommend you to install Debian Jessie as base system for IvozProvider.

You can also use one of the auto-install cds (see the project readme) that should also be working.

Thanks!

**Comment by chiragd** [comment]
@Kaian  Thanks for the quick response.

Is it possible to get a version that runs on ARM so I can test on a raspberry pi?

If you could point me in the right direction I could try to make a version of ivozprovider that would run on ubuntu.

On the RPi this is the error I get:

W: Failed to fetch http://packages.irontec.com/debian/dists/oasis/InRelease  Unable to find expected entry 'main/binary-armhf/Packages' in Release file (Wrong sources.list entry or malformed file)

W: Failed to fetch http://packages.irontec.com/debian/dists/chloe/InRelease  Unable to find expected entry 'main/binary-armhf/Packages' in Release file (Wrong sources.list entry or malformed file)

E: Some index files failed to download. They have been ignored, or old ones used instead.

thanks again in advance,

C

**Comment by Kaian** [comment]
Hi @chiragd 

Ivozprovider is a solution designed for handling lots of concurrents calls, so there is some minimum hardware requirements that must be fulfilled. There is a section in the documentation about [minimum requirements](https://irontec.github.io/ivozprovider/en/installation/requirements.html) even for the standalone version. While some components (like the media-relays) can be installed in small systems, the platform itself won't be suitable for a RPi.

Our debian packages repositories for ivozprovider only have amd64 and i386 architectures. As the warning display, you're trying to fetch _armhf_ packages that are not available. In the future, we maybe include some packages (like the mentioned relays) for that architecture, but not the whole suite.

All our installations use Debian Jessie as base system and only have ivozprovider installed on them. Our packages changes multiple systems files (from dns resolution to php settings) in order to run properly, so other softwares may not work at the same time or even break ivozprovider.

If you want to give a try, we highly recommend to install a virtual machine with an install-cd.

We're not really interested in adding support for Ubuntu. We prefer to spend our efforts in improving the solution and fixing problems, but in case you still want to build ivozprovider packages for ubuntu, our project (and [the forks of the other softwares](https://github.com/ivozprovider)) have a branch called oasis (the name of the release) with a debian folder that is used to create debian files. You can find manuals to build debian packages (using dpkg-builpackage) if you want to build them, but as mentioned we discourage it.

Thanks for all the feedback!!

**Comment by chiragd** [comment]
Hi @Kaian 

I got it running as a VM.

Why doesn't ivozprovider give access to the default asterisk sounds? Also some sample locutions would have been a nice touch.

So far the platform seems really cool. I'd love to give you feedback. What's the best way to do that?

**Comment by Kaian** [comment]
Nice! Thanks for taking the effort to test it!

Locutions are designed to customize some companies objects behaviour. We want them to upload their own sounds (aka locutions), what allows us to use and manipulate our sounds (asterisk ones) freely. Some **plaftorm admins** may want to replace all asterisk sounds for better ones recoded in a sound studio. We also try to make the plataform agnostic to the software: neither brand nor companies admins access an asterisk table with its data. They manipulate users, extensions, but they didn't need to know they are running asterisk, what channel driver, or anything specific to asterisk.

Locutions should be optional everywhere (IVR has a required welcome locution but will be make optional in the future)

About the demo data, we created the smallest possible dataset to make easier the proccess of testing a call between two users. We could create one of each objects (ivr, conferences, etc), but we wanted something very simple and documment the rest of the available objects (check the docs!).  Maybe in the future we change our minds, but somehow we prefered simplicity for the initial demo data.

We can not provide demo data in filesystem, just database entries due to the way the debian packages work: It will keep deploying demo files even if we delete the whole demobrand. In some sections (provisioning templates, invoices, etc) there are links in the help section with some samples you can download.

You can use github issues or the user mailing list to provide feedback and suggestions, they are highly appreciated. Users, devs and even commercial support contact emails are available in the project front page.

Thans for the testing!

**Comment by chiragd** [comment]
If asterisk sounds were installed it would have been nice to be able to use some of them like "the user is currently busy"

Also the UI doesn't show any music on hold options. When i make a test call music on hold is played but there is no option to select.

I wasn't talking about demo sounds for the demo company. I'm talking about generic sample sounds that could be used by any company as a starting point. Then if they want to they can record there own and upload them.

It would also be cool if there was a way to generate a call from the UI to a user, the audio would be reocrsd and saved as a sound file to be used as a locution. Or an access code like *99 that allows users to record audio and then have it available as a locution.

If i have machines running Kamailio and asterisk already, can use parts of the solution(like the UI) around my existing setup? I understand i would need to change my asterisk dialplan, install the agi and change the Kamailio config.

Also, why do you run two instances of kamailio for trunk and users? Could the same functionality not be achieved by one?

**Comment by Kaian** [comment]
Music on Hold can be specified at brand and company level. You have more info in [the docs](https://irontec.github.io/ivozprovider/en/pbx_features/music_on_hold.html). If neither brand nor company has music on hold configured, the generic platform music will be played (platform administrators can set their prefered moh in the filesystem, not through the portal interface).

Generic locutions could be a better approach that using asterisk one's, but we should make multiple's levels: global, brand and company, like music on holds right now. Could be interesting.

There's no click to dial, nor a way to make a phone ring right now in the portals interface, but we have some developments in mind to include a webrtc capable phone in userweb to archive this kind of behaviour. I like the idea of creating the locution without file, with a field like you mention *99<idlocution> so you can call with your phone and record (although this should only be allowed for locutions without file). 

You can use whatever you want from the project. It's free (as in freedom) software. Sadly the UI only has [documentation](http://irontec.github.io/klear-tutorial/) in spanish, but as mentioned before, feel free to request [comercial support](comercial@irontec.com) if you feel the need. You can take any part of the configuration, copy it, use for whatever you want (really, whatever!), but we're here to provide support on ivozprovider only :)

We use two kamailio because they have different task. As mentioned in my first note, this solution is designed to be installed distributed between multiples machines (standalone is actualy for testing the product), so both kamailio will run separately allowing them to handle more request at the same time from companies (user) and providers (trunks).

Thanks for the feedback

**Comment by Kaian** [comment]
In order to maintain a bit of order in github issues, I think we should close this as the topic is not longer related with the issue title.

Feel free to open new issues with questions on specific feedback for each section. Or if you prefer you can use the users mailing list so we can share it with other subscribed users.

Regards!

**Comment by chiragd** [comment]
Thanks for all the help. Will close now.
